docs/automated-testing/performance-testing/load-testing.md

```diff
@@ -26,21 +26,49 @@ There are a number of basic component which are required to carry out a load tes
+
+If a custom client is used, code development will be required:
+
+1. **Custom development** - Design for minimal impact/overhead.  Be sure to capture  only those features of the production client that are relevant from a load perspective.   Does it matter if the same test is duplicated, or must the workload be unique for each test?  Can all tests be run under the same user context?
+2. **Test environment** - Create test environment that resembles production environment.  This includes the platform as well as external systems, e.g., data sources.
+3. **Security contexts** - Be sure to have all requisite security contexts for the test environment.  Automation like pipelines may require special setup, e.g., OAuth2 client credential flow instead of auth code flow, because interactive login is replaced by non-interactive.  Allow planning leeway in case admin approval is required for new security contexts.
+4. **Test data strategy** - Make sure that output data format (ascii/binary/...) is compatible with whatever analysis tool is used in the analysis phase.  This also includes storage areas (local/cloud/...), which may trigger new security contexts.  Bear in mind that it may be necessary to collect data from sources external to the application to correlate potential performance issues with the application behavior.  This includes platform and network metrics.  Make sure to collect data that covers analysis needs (statistical measures, distributions, graphs, etc.).
+5. **Automation** -  Repeatability is critical.  It must be possible to re-run a given test multiple times to verify consistency and resilience of the application itself and the underlying platform.  Pipelines are recommended whenever possible.
+Evaluate whether load tests should be run as part of the PR strategy.
+6. **Test client debugging** - All test modules should be carefully debugged to ensure that the execution phase progresses smoothly.
+7. **Test client validation** - All test modules should be validated for extreme values of the input parameters.  This reduces the risk of running into unexpected difficulties when stepping through the full test matrix during the execution phase.
 
 ### Execution
 
-It is recommended to use an existing testing framework (see below). These tools will provide a method of both specifying the user activity scenarios and how to execute those at load. It is common to slowly ramp up to your desired load to better replicate real world behavior. Once you have reached your defined workload, maintain this level long enough to see if your system stabilizes. To finish up the test you should also ramp to see record how the system slows down as well.
+It is recommended to use an existing testing framework (see below). These tools will provide a method of both specifying the user activity scenarios and how to execute those at load.  Depending on the situation, it may be advisable to coordinate testing activities with the platform operations team.  
```

Comment:
```suggestion
It is recommended to use an existing testing framework (see below). These tools will provide a method of both specifying the user activity scenarios and how to execute those at load. Depending on the situation, it may be advisable to coordinate testing activities with the platform operations team.  
```